### ISSUE TITLE
NaN loss for 8bit

### ISSUE THREAD

**Issue by shehel** [yhhhli/APoT_Quantization#8]
hi, I was trying to train resnet18 on Imagenet with 8bit quantization. However, the loss is always Nan. Tried smaller learning rates and the results are the same. Note I tried running with batch size of 64 and 128 only. Although its low, it seemed to work with 5 bit quantization training. Any ideas why this could be?

**Comment by yhhhli** [comment]
Hi, 
If you are going to use 8-bit quantization, we recommend you directly quantize the whole network without re-training. You can write a quantization function just like we did in the first conv layer where we use 8-bit quantization. And you can run inference and find the loss degradation could be within 0.3%.
Regarding the 8-bit training NaN loss, I think you use the APoT scheme for the quantization, however, we do not implement the 8-bit for the APoT scheme. As you can see in the build_pot_values() function, we do not implement bitwidth higher than 5.
